Ni mulitple devices

Skoricius opened the pull request:
It is possible to have multiple NI devices that are working together and are synchronized (e.g. see [configurations with RTSI](https://www.ni.com/en-gb/support/documentation/supplemental/18/real-time-system-integration--rtsi--and-configuration-explained.html)).

However, in this case, the current Task class will not work because channels are listed purely by names meaning that the two different channels on separate devices can have the same name, so "Duplicate channel name..." exception will be raised. Also, in addition to the channel type, the mini tasks need to be separated for each device.
To preserve backwards compatibility, I created a separate class which is very similar to the existing Task class as this TaskMultiDevice requires full path to the output devices when writing data:
```python
dev1 = NIDAQ('Dev1')
dev2 = NIDAQ('Dev2')
task = TaskMultiDevice(dev1.ao2, dev2.ao2, dev1.ai23)
task.set_timing(duration='1s', fsamp='10Hz')
write_data = {'Dev1/ao2': np.linspace(0, 9, 10) * u.V, 'Dev2/ao2': np.linspace(0, 9, 10) * u.V}
result = task.run(write_data)
```
If you are OK with that, I could do the modifications directly on the Task as well because everything else should work the same way.

natezb commented:
Thanks for this! I haven't looked over the code in depth, but it does look like there's some pretty significant duplication of `Task` code. So, I think some other solution is likely better: probably (as you suggested) modifying `Task`, or perhaps some other approach, e.g. making `TaskMultiDevice` use `Task`.

How would the `Task` interface have to change to accommodate these modifications (i.e. what existing code would break)? If it's fairly minimal, I'm leaning towards just modifying `Task` directly.

Skoricius commented:
Yes, as you said, a lot of the code was taken directly from the task. I agree, that makes more sense and I have now merged them.

The only difference in the interface is that now, instead of defining channels by name when writing data, they need to be defined by their full path (e.g. `Dev1/ao2` instead of `ao2`, see the code snippet in my first comment).

Two more minor things unrelated to the multiple device issue that I added:
1. Added `SetWriteRegenMode` method in the NiceObject task. It's not used in the code directly, but I needed that method for my project and it was not defined. I think it's useful to have.
2. Code now waits until the task is done before stopping in `Task.run()` method (line 590). This is important because if there are no AI channels defined, the `read` method instantly returns and the program tries to stop the task as soon as it has started.

natezb commented:
Great, thanks for contributing!
